components/user_settings/advanced/user_settings_advanced.tsx

```diff
@@ -450,9 +450,9 @@ export default class AdvancedSettingsDisplay extends React.PureComponent<Props,
                         </fieldset>,
                     ]}
                     setting={Preferences.UNREAD_SCROLL_POSITION}
-                    submit={this.handleSubmit}
```

Comment:
why was this renamed?

Reply:
submit handle many kinds of callback which have different signature. I had been working ok until migration. 

 In order to handle this kind of situation, I think it is better to separate two functions one is `submit?: null | ((setting?: string) => void) | (() => Promise<void>);` the other is this `submitWithSettings?: (settings: string[]) => void;`

Reply:
i see but we need to change the tests to accomodate these changes. i suggest spending a bit more time to see if we can work with just `submit` and if not then let me know we can work something out

Reply:
I got it. thanks a lot

Reply:
I am the same opinion. We should keep the name and change the function signature instead. we now have two separate props that essentially do the same, but with different parameters. Could you please adjust the signature on all occurences to accept a `string[]` instead?

Reply:
ok but I should expand my changes for this task. I'm afraid that my changes are getting bigger and bigger. 

If you don't mind for that I would try